packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -280,11 +344,20 @@ function unmount(vnode: VNode, parent: ParentNode, doRemove: boolean = false) {
 
     // When unmounting a VNode subtree not all the elements have to removed from the DOM. The
     // subtree root, is the only element worth unmounting from the subtree.
-    if (doRemove) {
-        removeNode(elm!, parent);
+    // elm may be undefined for light dom slots
+    if (doRemove && elm) {
+        removeNode(elm, parent);
     }
 
     switch (type) {
+        case VNodeType.Slot:
+            unmountVNodes(
+                isUndefined(vnode.aChildren) ? vnode.children : vnode.aChildren.vnodes,
+                (elm as ParentNode) ?? parent,
+                doRemove
```

Comment:
The `doRemove` flag shouldn't be passed down, only the top-level element should be unmounted.